ironfish-cli/src/utils/transaction.ts

```diff
@@ -488,8 +491,8 @@ export async function watchTransaction(options: {
   let currentStatus = prevStatus
 
   // If the transaction is already in the desired state, return
-  if (currentStatus === waitUntil) {
-    logger.log(`Transaction ${options.hash} is ${waitUntil}`)
+  if (currentStatus !== 'not found' && waitUntil.includes(currentStatus)) {
```

Comment:
It was a weird type issue because of the assignment above. It was introduced here: https://github.com/iron-fish/ironfish/pull/3726/files . Probably not the best initial implementation tbh. I can look into fixing it but might be out of scope of this PR?